File: ECommarce.View/src/app/core/models/product.ts
export interface ProductImage {
  id: number;
  imageUrl: string;
  altText?: string;
  isPrimary: boolean;
}

export interface ProductVariant {
  id: number;
  sku?: string;
  size?: string;
  price?: number;
  compareAtPrice?: number;
  purchaseRate?: number;
  stockQuantity: number;
  isDefault?: boolean;
}

export enum ProductType {
  Simple = 0,
}

export interface Product {
  id: number;
  name: string;
  slug: string;
  description?: string;
  shortDescription?: string;
  sku: string;
  price: number;
  compareAtPrice?: number;
  purchaseRate?: number;
  stockQuantity: number;
  isActive: boolean;
  isNew: boolean;
  isFeatured: boolean;

  categoryId: number;
  categoryName: string;
  subCategoryId?: number;
  subCategoryName?: string;
  collectionId?: number;
  collectionName?: string;

  imageUrl?: string;
  images: ProductImage[];
  variants: ProductVariant[];

  metaTitle?: string;
  metaDescription?: string;
  fabricAndCare?: string;
  shippingAndReturns?: string;
  sizeChartUrl?: string;

  // ilyn.global Design Fields
  tier?: string;
  tags?: string;
  sortOrder?: number;

  productType: ProductType;
}

export interface RelatedProduct {
  id: number;
  name: string;
  sku?: string;
  price: number;
  compareAtPrice?: number;
  imageUrl: string;
  slug: string;
  tier?: string;
  variants?: ProductVariant[];
}

File: ECommerce.Core/Entities/Product.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using ECommerce.Core.Enums;

namespace ECommerce.Core.Entities;

public class Product : BaseEntity
{
    public string Name { get; set; } = string.Empty;
    public string Slug { get; set; } = string.Empty;
    public string? Description { get; set; }
    public string? ShortDescription { get; set; }
    public string? Sku { get; set; }
    public string? ImageUrl { get; set; }
    public int StockQuantity { get; set; }
    public bool IsActive { get; set; } = true;
    public ProductType ProductType { get; set; } = ProductType.Simple;

    public bool IsNew { get; set; } = false;
    public bool IsFeatured { get; set; } = false;
    
    // Meta Info
    public string? MetaTitle { get; set; }
    public string? MetaDescription { get; set; }
    public string? FabricAndCare { get; set; }
    public string? ShippingAndReturns { get; set; }
    public string? SizeChartUrl { get; set; }
    
    // New Fields
    public string? Tier { get; set; }
    public string? Tags { get; set; }
    public int SortOrder { get; set; }

    // Relations
    public int CategoryId { get; set; }
    public Category? Category { get; set; }

    public int? SubCategoryId { get; set; }
    public SubCategory? SubCategory { get; set; }

    public int? CollectionId { get; set; }
    public Collection? Collection { get; set; }

    public ICollection<ProductVariant> Variants { get; set; } = new List<ProductVariant>();
    public ICollection<ProductImage> Images { get; set; } = new List<ProductImage>();
    public ICollection<Review> Reviews { get; set; } = new List<Review>();

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? UpdatedAt { get; set; }
}
